zeroize_derive: use of PhantomData

I currently try to apply the `zeroize` v1.5 with the `zeroize_derive` feature.

I would like to use the `zeroize_derive` feature for a stuct which contains PhantomData, [for example](https://github.com/aewag/utils/blob/4f34fd7b4c37950ee06c89650ec620789bd0826a/zeroize/tests/zeroize_derive.rs#L32).

This fails to compile and displays following error message:

```
$ cargo test --features zeroize_derive
   Compiling zeroize v1.5.1 (/home/alwagner/working_directory/projects/research/rust_crypto/utils/zeroize)
WARN rustc_metadata::locator no metadata found: incompatible metadata version found: '/home/alwagner/working_directory/projects/research/rust_crypto/utils/target/debug/deps/libzeroize_derive-b9ca4201b79543dd.so'
error[E0034]: multiple applicable items in scope
  --> zeroize/tests/zeroize_derive.rs:20:27
   |
20 |         #[derive(Zeroize, ZeroizeOnDrop)]
   |                           ^^^^^^^^^^^^^ multiple `zeroize_or_on_drop` found
   |
   = note: candidate #1 is defined in an impl of the trait `AssertZeroizeOnDrop` for the type `&mut T`
   = note: candidate #2 is defined in an impl of the trait `AssertZeroize` for the type `T`
   = note: this error originates in the derive macro `ZeroizeOnDrop` (in Nightly builds, run with -Z macro-backtrace for more info)
help: disambiguate the associated function for candidate #1
   |
20 |         #[derive(Zeroize, AssertZeroizeOnDrop::zeroize_or_on_drop(&mut ZeroizeOnDrop))]
   |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the associated function for candidate #2
   |
20 |         #[derive(Zeroize, AssertZeroize::zeroize_or_on_drop(&mut ZeroizeOnDrop))]
   |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0034]: multiple applicable items in scope
  --> zeroize/tests/zeroize_derive.rs:25:27
   |
25 |         #[derive(Zeroize, ZeroizeOnDrop)]
   |                           ^^^^^^^^^^^^^ multiple `zeroize_or_on_drop` found
   |
   = note: candidate #1 is defined in an impl of the trait `AssertZeroizeOnDrop` for the type `&mut T`
   = note: candidate #2 is defined in an impl of the trait `AssertZeroize` for the type `T`
   = note: this error originates in the derive macro `ZeroizeOnDrop` (in Nightly builds, run with -Z macro-backtrace for more info)
help: disambiguate the associated function for candidate #1
   |
25 |         #[derive(Zeroize, AssertZeroizeOnDrop::zeroize_or_on_drop(&mut ZeroizeOnDrop))]
   |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the associated function for candidate #2
   |
25 |         #[derive(Zeroize, AssertZeroize::zeroize_or_on_drop(&mut ZeroizeOnDrop))]
   |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0034`.
error: could not compile `zeroize` due to 2 previous errors
```

Shall this even be supported or am I misusing this feature? (FYI: I created a [draft PR](https://github.com/aewag/utils/pull/4) on my fork to test a bit.)

--- a/zeroize/src/lib.rs
+++ b/zeroize/src/lib.rs
@@ -271,7 +271,7 @@ pub mod __internal {
         fn zeroize_or_on_drop(self);
     }
 
-    impl<T: ZeroizeOnDrop> AssertZeroizeOnDrop for &mut T {
+    impl<T: ZeroizeOnDrop> AssertZeroizeOnDrop for &&mut T {
         fn zeroize_or_on_drop(self) {}
     }
 

--- a/zeroize/tests/zeroize_derive.rs
+++ b/zeroize/tests/zeroize_derive.rs
@@ -242,11 +242,41 @@ mod custom_derive_tests {
     }
 
     #[test]
-    fn derive_only_zeroize_on_drop() {
+    fn derive_inherit_zeroize_on_drop() {
         #[derive(ZeroizeOnDrop)]
         struct X([u8; 3]);
 
         #[derive(ZeroizeOnDrop)]
+        struct Z(X);
+
+        let mut value = Z(X([1, 2, 3]));
+        unsafe {
+            std::ptr::drop_in_place(&mut value);
+        }
+        assert_eq!(&value.0 .0, &[0, 0, 0])
+    }
+
+    #[test]
+    fn derive_inherit_from_both() {
+        #[derive(Zeroize, ZeroizeOnDrop)]
+        struct X([u8; 3]);
+
+        #[derive(ZeroizeOnDrop)]
+        struct Z(X);
+
+        let mut value = Z(X([1, 2, 3]));
+        unsafe {
+            std::ptr::drop_in_place(&mut value);
+        }
+        assert_eq!(&value.0 .0, &[0, 0, 0])
+    }
+
+    #[test]
+    fn derive_inherit_both() {
+        #[derive(Zeroize, ZeroizeOnDrop)]
+        struct X([u8; 3]);
+
+        #[derive(Zeroize, ZeroizeOnDrop)]
         struct Z(X);
 
         let mut value = Z(X([1, 2, 3]));

